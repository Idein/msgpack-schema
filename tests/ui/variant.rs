use msgpack_schema::*;

mod serialize {
    use super::*;

    #[derive(Serialize)]
    enum E1 {
        #[optional]
        V1,
    }

    #[derive(Serialize)]
    enum E2 {
        #[untagged]
        V1,
    }

    #[derive(Serialize)]
    enum E3 {
        #[tag = 1]
        V1(#[untagged] String),
    }

    #[derive(Serialize)]
    enum E4 {
        #[tag = 1]
        V1(#[optional] String),
    }

    #[derive(Serialize)]
    enum E5 {
        #[tag = 1]
        V1(#[tag = 1] String),
    }

    #[derive(Serialize)]
    #[untagged]
    enum E6 {
        #[optional]
        V1,
    }

    #[derive(Serialize)]
    #[untagged]
    enum E7 {
        #[untagged]
        V1,
    }

    #[derive(Serialize)]
    #[untagged]
    enum E8 {
        #[tag = 1]
        V1,
    }

    #[derive(Serialize)]
    #[untagged]
    enum E9 {
        V1(#[untagged] String),
    }

    #[derive(Serialize)]
    #[untagged]
    enum E10 {
        V1(#[optional] String),
    }

    #[derive(Serialize)]
    #[untagged]
    enum E11 {
        V1(#[tag = 1] String),
    }
}

mod deserialize {
    use super::*;

    #[derive(Deserialize)]
    enum E1 {
        #[optional]
        V1,
    }

    #[derive(Deserialize)]
    enum E2 {
        #[untagged]
        V1,
    }

    #[derive(Deserialize)]
    enum E3 {
        #[tag = 1]
        V1(#[untagged] String),
    }

    #[derive(Deserialize)]
    enum E4 {
        #[tag = 1]
        V1(#[optional] String),
    }

    #[derive(Deserialize)]
    enum E5 {
        #[tag = 1]
        V1(#[tag = 1] String),
    }

    #[derive(Deserialize)]
    #[untagged]
    enum E6 {
        #[optional]
        V1,
    }

    #[derive(Deserialize)]
    #[untagged]
    enum E7 {
        #[untagged]
        V1,
    }

    #[derive(Deserialize)]
    #[untagged]
    enum E8 {
        #[tag = 1]
        V1,
    }

    #[derive(Deserialize)]
    #[untagged]
    enum E9 {
        V1(#[untagged] String),
    }

    #[derive(Deserialize)]
    #[untagged]
    enum E10 {
        V1(#[optional] String),
    }

    #[derive(Deserialize)]
    #[untagged]
    enum E11 {
        V1(#[tag = 1] String),
    }
}

fn main() {}
