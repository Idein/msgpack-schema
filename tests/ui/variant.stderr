error: #[optional] at an invalid position
 --> $DIR/variant.rs:8:9
  |
8 |         #[optional]
  |         ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/variant.rs:14:9
   |
14 |         #[untagged]
   |         ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/variant.rs:21:12
   |
21 |         V1(#[untagged] String),
   |            ^^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/variant.rs:27:12
   |
27 |         V1(#[optional] String),
   |            ^^^^^^^^^^^

error: #[tag] at an invalid position
  --> $DIR/variant.rs:33:12
   |
33 |         V1(#[tag = 1] String),
   |            ^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/variant.rs:39:9
   |
39 |         #[optional]
   |         ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/variant.rs:46:9
   |
46 |         #[untagged]
   |         ^^^^^^^^^^^

error: #[tag] at an invalid position
  --> $DIR/variant.rs:53:9
   |
53 |         #[tag = 1]
   |         ^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/variant.rs:60:12
   |
60 |         V1(#[untagged] String),
   |            ^^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/variant.rs:66:12
   |
66 |         V1(#[optional] String),
   |            ^^^^^^^^^^^

error: #[tag] at an invalid position
  --> $DIR/variant.rs:72:12
   |
72 |         V1(#[tag = 1] String),
   |            ^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/variant.rs:81:9
   |
81 |         #[optional]
   |         ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/variant.rs:87:9
   |
87 |         #[untagged]
   |         ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/variant.rs:94:12
   |
94 |         V1(#[untagged] String),
   |            ^^^^^^^^^^^

error: #[optional] at an invalid position
   --> $DIR/variant.rs:100:12
    |
100 |         V1(#[optional] String),
    |            ^^^^^^^^^^^

error: #[tag] at an invalid position
   --> $DIR/variant.rs:106:12
    |
106 |         V1(#[tag = 1] String),
    |            ^^^^^^^^^^

error: #[optional] at an invalid position
   --> $DIR/variant.rs:112:9
    |
112 |         #[optional]
    |         ^^^^^^^^^^^

error: #[untagged] at an invalid position
   --> $DIR/variant.rs:119:9
    |
119 |         #[untagged]
    |         ^^^^^^^^^^^

error: #[tag] at an invalid position
   --> $DIR/variant.rs:126:9
    |
126 |         #[tag = 1]
    |         ^^^^^^^^^^

error: #[untagged] at an invalid position
   --> $DIR/variant.rs:133:12
    |
133 |         V1(#[untagged] String),
    |            ^^^^^^^^^^^

error: #[optional] at an invalid position
   --> $DIR/variant.rs:139:12
    |
139 |         V1(#[optional] String),
    |            ^^^^^^^^^^^

error: #[tag] at an invalid position
   --> $DIR/variant.rs:145:12
    |
145 |         V1(#[tag = 1] String),
    |            ^^^^^^^^^^
